lib/legacy-gen/v1generator.js

```diff
@@ -0,0 +1,19 @@
+function generateV1Config (platformConfig) {
```

Comment:
Both this and legacy could be implemented as transitions on the current format such that the code would just recursively call it and then there'd be less versioned code over time. v0 could special case the old files perhaps too.

```js
export const legacyFormats = [
   {
        configVersion: 'v1',
        transition(dataIn) {
            const dataOut = dataIn;
            ...
            return dataOut
        },
    }
    {
        configVersion: 'v0',
        ....
    }
]
```

```js
import { legacyFormats } from './legacyFormats.js'

function writeToDisk(currentFormat) {
    writeConfigToDisk(currentConfigVersion, platform, dataOut)
    for (let format of legacyFormats) {
        const dataOut = format.transition(currentFormat)
        if (format.configVersion === 'v0') {
            writeLegacyFilesToDisk(dataOut)
        } else {
            writeConfigToDisk(format.configVersion, platform, dataOut)
        }
        currentFormat = dataOut
    }
}
```

Reply:
I like the idea of the legacy transition structure but I think the v0 config files are just special enough, especially with the need for the unprotectedTemp object, that they should be kept as a separate case done at then end of generation. The others make sense to be done per platform file on write but the v0 files only need to be done once.